beacon_node/eth2-libp2p/src/peer_manager/mod.rs

```diff
@@ -263,6 +263,53 @@ impl<TSpec: EthSpec> PeerManager<TSpec> {
         }
     }
 
+    pub fn handle_rpc_error(&mut self, peer_id: &PeerId, protocol: Protocol, err: &RPCError) {
+        // map this error with a `PeerAction` asignated to the peer (if any), and decide if the
+        // peer manager should request some `PeerManagerEvent`
+        let (peer_action, event) = match err {
+            RPCError::IncompleteStream => {
+                // TODO: InvalidMessage? they Clogged? offline?
+                (None, None)
+            }
+            RPCError::InternalError(reason) => {
+                // NOTE: our fault. Now they hate us. Do nothing
+                (None, None)
+            }
+            RPCError::InvalidData => {
+                // TODO: invalid? malicious?
+                (None, None)
+            }
+            RPCError::IoError(e) => {
+                // TODO Whose fault? do nothing?
+                (None, None)
+            }
+            RPCError::RPCErrorResponse => {
+                // TODO what does this mean?
```

Comment:
From the RPC types, there is an RPCErrorResponse(Success/Failure)... I imagine this is the case where we got an RPCErrorResponse(Failure) IIRC the type names correctly. 
This happens when a client reject a request from us and gives us a reason. There are three kinds of enums and a custom string. 
If the peer has kicked us, depending on the reason (for example Prysm could have rate limited this request). 
We probably need to think about the reasons the request failed here and deal with them accordingly.
I'll let you have a look at the types, and see what you think we should do